### ISSUE TITLE
SQLFluff does not recognize trailing commas in select statements in BigQuery

### ISSUE THREAD

**Issue by ericxiao251** [sqlfluff/sqlfluff#389]
In BigQuery it is legal to have a trailing comma in the last column in a select statement ie.

```sql
select
	1,
	2,
	3,
from test
```

in normal ANSII SQL the comma after 3 is illegal, but in BigQuery it is.

when you run sqlfluff against this, you get a "Found unparsable section":
```sql
== [models/ml_projects/app_revenue_weekly_usd/app_revenue_weekly_usd.sql] FAIL
L:   1 | P:   1 |  PRS | Found unparsable section: 'SELECT\n api_client_id AS
                       | api_client_id,...'
```

**Comment by alanmcruickshank** [comment]
I thought I had implemented this 🤔 .

`allow_trailing` is definitely there in the dialect: https://github.com/alanmcruickshank/sqlfluff/blob/master/src/sqlfluff/dialects/dialect_ansi.py#L590

What version are you running on?

**Comment by ericxiao251** [comment]
DBT SQL:

```sql
{{ config(materialized='view') }}

SELECT
  invoice_transaction_at,
  DATE(TIMESTAMP_TRUNC(invoice_transaction_at, week)) AS start_date_of_week,

FROM {{ source('finance', 'reconciled_cash_facts') }}
```

DBT run:
```shell
~... ⌚ 11:09:24
$ bin/dbt run --model app_revenue_weekly_usd  --target dev
Running with dbt=0.16.0
...

11:09:14 | Concurrency: 8 threads (target='dev')
11:09:14 |
11:09:14 | 1 of 1 START view model dbt_dev_eric.app_revenue_weekly_usd.......... [RUN]
11:09:15 | 1 of 1 OK created view model dbt_dev_eric.app_revenue_weekly_usd..... [CREATE VIEW in 1.15s]
11:09:15 |
11:09:15 | Finished running 1 view model in 4.22s.

Completed successfully

Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

SQLFluff lint run with `-vv`:
```shell
~... ⌚ 11:09:24
$ sqlfluff lint models/app_revenue_weekly_usd.sql -vv --exclude-rules L027,L016
==== sqlfluff ====
sqlfluff:           0.3.4 python:             3.8.0
dialect:         bigquery verbosity:              2
== Raw Config:
core:
    dialect:            bigquery
    exclude_rules:      L027,L016
    ignore:             []
    nocolor:            False
    output_line_length: 80
    recurse:            True
    templater:          jinja
    verbose:            2
indentation:
    indented_joins:     False
rules:
    indent_unit:        space
    max_line_length:    80
    tab_space_size:     2
    L010:
        capitalisation_policy:upper
    L019:
        comma_style:        trailing
templater:
    jinja:
        apply_dbt_builtins: True
        macros:
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable) %}item{% endmacro %}

==== readout ====

=== [ path: models/app_revenue_weekly_usd.sql ] ===

== [models/app_revenue_weekly_usd.sql] PARSING
== [models/app_revenue_weekly_usd.sql] FAIL
L:   3 | P:   1 |  PRS | Found unparsable section: 'SELECT\n
                       | invoice_transaction_at,\n DATE(...'

==== summary ====
files:             1 violations:        1
clean files:       0 unclean files:     1
avg per file:   1.00 unclean rate:   100%
status:         FAIL
```

SQLFluff version:
```shell
~... ⌚ 11:09:24
$ pip freeze | grep sqlfluff
sqlfluff==0.3.4
```

**Comment by alanmcruickshank** [comment]
I've tested this on `0.3.4` and can confirm it fails, and I've tested it on `0.3.5` and `0.3.6` and can confirm it's fixed. 👍 

I'm going to close this issue.
